apps/desktop/src/app/sync/sync.service.ts

```diff
@@ -204,7 +205,36 @@ export class SyncService {
   private watchProcessing() {
     return this.sdk.nuclia.db
       .pull()
-      .pipe(repeatWhen((obs) => obs.pipe(delay(10000))))
+      .pipe(
+        repeatWhen((obs) => obs.pipe(delay(10000))),
+        filter((res: ProcessingPullResponse) => res.status !== 'empty' && !!res.payload),
+        map((res) => res.payload as string),
+        switchMap((base64) => NucliaProtobufConverter(convertDataURIToBinary(base64))),
+        map((data: any) => {
+          const sync = this._queue.find((sync) => sync.fileUUIDs.includes(data.uuid));
+          let item: SyncItem | undefined;
+          if (sync) {
+            item = sync.files.find((file) => file.uuid === data.uuid);
+          }
+          return { data, sync, item } as { data: any; sync: Sync | null; item?: SyncItem };
+        }),
+        filter(({ sync, item }) => !!sync && !!item),
+        map(({ data, sync, item }) => ({ data, sync, item } as { data: any; sync: Sync; item: SyncItem })),
```

Comment:
if you reverse these 3 steps, you do not need `if` anymore:
```ts
map((data: any) => ({ data, sync: this._queue.find((sync) => sync.fileUUIDs.includes(data.uuid)) })),
filter(({data, sync}) => !!sync),
map(({data, sync}) => ({ data, sync, item: sync.files.find((file) => file.uuid === data.uuid) })),
filter(({data, sync, item}) => !!item),
```
not a huge improvement though, more a matter of taste